### PR TITLE
mark Lightstep provider deprecated

### PR DESCRIPTION
**Pull request by zirain**
xref: https://github.com/istio/istio/issues/40027

### CODE REVIEW

File: mesh/v1alpha1/config.proto [hunk 1]
```diff
@@ -487,10 +487,8 @@ message MeshConfig {
       // Configures a tracing provider that uses the Zipkin API.
       ZipkinTracingProvider zipkin = 4;
       // Configures a Lightstep tracing provider.
-      // Note: For Istio 1.15+, configuring this provider will result in
-      // using an OpenTelemetryTracingProvider configured specially for 
-      // Lightstep. This is part of the Lightstep transition to OpenTelemetry.
-      LightstepTracingProvider lightstep = 5;
+      // Deprecated: For Istio 1.15+, please use an OpenTelemetryTracingProvider instead, more details can be found at https://github.com/istio/istio/issues/40027
+      LightstepTracingProvider lightstep = 5 [deprecated=true];
```
Comment (hzxuzhonghu) [inline mesh/v1alpha1/config.proto:491]: need to hide from doc

Comment (zirain) [reply mesh/v1alpha1/config.proto:491]: I'd like to mark as deprecated in 1.17, then hided from doc in 1.18.

Comment (hzxuzhonghu) [reply mesh/v1alpha1/config.proto:491]: We usually make them together

Comment (zirain) [reply mesh/v1alpha1/config.proto:492]: updated

**Review by hzxuzhonghu — APPROVED** [review]

**Comment by ericvn** [comment]
Should there be a release note if we are deprecating? Maybe it will be elsewhere?

**Comment by zirain** [comment]
> Should there be a release note if we are deprecating? Maybe it will be elsewhere?

updated

**Review by linsun — APPROVED** [review]
Thx for adding the release note

**Pull request merged**